src/cmdline/CommandLine.cpp

```diff
@@ -0,0 +1,579 @@
+
+        return true;
+    }
+
+    static bool ParseShortOption(const CommandLineOptionDefinition * options,
+                                 CommandLineArgEnumerator *argEnumerator,
+                                 const char *argument)
+    {
+        const CommandLineOptionDefinition * option = nullptr;
+
+        const char * shortOption = &argument[1];
+        for (; *shortOption != '\0'; shortOption++)
+        {
+            option = FindOption(options, shortOption[0]);
+            if (option == nullptr)
+            {
+                Console::WriteError("Unknown option: -");
+                Console::WriteError(shortOption[0]);
+                Console::WriteLineError();
+                return false;
+            }
+            if (option->Type == CMDLINE_TYPE_SWITCH)
+            {
+                if (!ParseOptionValue(option, nullptr))
+                {
+                    return false;
+                }
+            }
+            else if (shortOption[1] != '\0')
+            {
+                if (!ParseOptionValue(option, &shortOption[1]))
+                {
+                    return false;
+                }
+                return true;
+            }
+        }
+
+        if (option != nullptr && option->Type != CMDLINE_TYPE_SWITCH)
+        {
+            const char * valueString = nullptr;
+            if (!argEnumerator->TryPopString(&valueString))
+            {
+                Console::WriteError("Expected value for option: ");
+                Console::WriteError(option->ShortName);
+                Console::WriteLineError();
+                return false;
+            }
+                
+            if (!ParseOptionValue(option, valueString))
+            {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    static bool ParseOptionValue(const CommandLineOptionDefinition * option, const char * valueString)
+    {
+        if (option->OutAddress == nullptr) return true;
+
+        switch (option->Type) {
+        case CMDLINE_TYPE_SWITCH:
+            *((bool *)option->OutAddress) = true;
+            return true;
+        case CMDLINE_TYPE_INTEGER:
+            *((sint32 *)option->OutAddress) = (sint32)atol(valueString);
+            return true;
+        case CMDLINE_TYPE_REAL:
+            *((float *)option->OutAddress) = (float)atof(valueString);
+            return true;
+        case CMDLINE_TYPE_STRING:
+            *((utf8 * *)option->OutAddress) = String::Duplicate(valueString);
+            return true;
+        default:
+            Console::WriteError("Unknown CMDLINE_TYPE for: ");
+            Console::WriteLineError(option->LongName);
+            return false;
+        }
+    }
+
+    static bool HandleSpecialArgument(const char * argument)
+    {
+#if defined(__APPLE__) && defined(__MACH__)
+        if (String::Equals(argument, "-NSDocumentRevisionsDebugMode"))
+        {
+            return true;
+        }
+        if (String::StartsWith("-psn_"))
```

Comment:
"Too few arguments to function call, expected at least 2, have 1" I get this error when trying to build.